app/core/sandbox.py

```diff
@@ -0,0 +1,203 @@
+"""
+This module provides the Sandbox class with a set of methods to trigger and fetch workflow jobs and runs
+"""
+
+import json
+import logging
+
+from app.core.github import ClusterGithubClient
+from app.helpers.constants import (
+    WorkflowTypes,
+    WORKFLOW_ACTIVE,
+)
+from app.helpers.db_utils import DBSession
+from app.helpers.exceptions import ClusterWorkflowNotFoundException
+from app.helpers.utils import get_secret
+from app.models.request_models import Workflow, WorkflowJob
+from app.models.sql_models import CancelledRun
+
+logger = logging.getLogger(__name__)
+
+cluster_github_client = ClusterGithubClient(
+    get_secret("pr-sandbox-cluster-installation-id")
+)
+
+
+class Sandbox:
+    def __init__(self, sandbox_name: str):
+        self.sandbox_name = sandbox_name
+        self.workflows = cluster_github_client.get_workflow_list()
+        self.in_progress_workflow_runs = None
+
+    @property
+    def exists(self) -> bool:
+        """
+        Checks if the sandbox exists in cluster.
+        """
+        tutor_config = cluster_github_client.get_instance_config(self.sandbox_name)
+        return tutor_config is not None
+
+    def _get_workflow(self, type: WorkflowTypes) -> Workflow:
+        """
+        Fetch the workflow of the given type.
+
+        Args:
+            type (WorkflowTypes): The workflow type
+
+        Returns:
+            Workflow: Object with details of the matching workflow
+        """
+        for workflow in self.workflows:
+            if workflow.type == type and workflow.state == WORKFLOW_ACTIVE:
+                return workflow
```

Comment:
Is there a chance this returns `None`? I know we are in control of the code and type as well, though it would feel a bit better to have a `raise` to make sure we never caught off guard?


```suggestion
    def _get_workflow(self, workflow_type: WorkflowTypes) -> Workflow:
        """
        Fetch the workflow of the given type.

        Args:
            workflow_type (WorkflowTypes): The workflow type

        Returns:
            Workflow: Object with details of the matching workflow
        """
        for workflow in self.workflows:
            if workflow.type == workflow_type and workflow.state == WORKFLOW_ACTIVE:
                return workflow
        
        raise RuntimeError(f"Workflow type {workflow_type} does not exist or not active")
```